l10n_it_asset_history_import/wizards/asset_history_import.py

```diff
@@ -434,15 +454,29 @@ def convert_to_asset_vals(self, asset_dict, workbook, sheet):
                     "Is it in its ISO 4217 format?"
                 )
             )
+        supplier_vat_col = get_header_by_model_and_field("res.partner", "vat").col
+        supplier_vat = asset_dict[supplier_vat_col]
+        partner = self.env["res.partner"]
+        if supplier_vat:
+            partner = partner.search([("vat", "ilike", supplier_vat)])[:1]
```

Comment:
Might be better for performances to use `limit=1` as `search` parameter instead of `[:1]`.
Else, you can use `fields.first(partner.search([("vat", "ilike", supplier_vat)]))` which will also guarantee to maintain the recordset order.